### PR TITLE
[AEA-944] Fix a bug on 'aea push --local' with 'latest' as version

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

There was a problem in the comparison between the provided public id and the existing local package public id the 'aea push' CLI command, when the '--local' flag was provided and the public id provided as argument contained ':latest' (or equivalently '') as version number. Now if latest is provided, only the public id prefix is compared, o/w also the version is compared.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: tests/test_cli/test_push.py [hunk 1]
```diff
@@ -207,3 +211,31 @@ def test_push_contract_registry_positive(self, *mocks):
...
+
+
+class TestPushLocallyWithLatest(AEATestCaseEmpty):
+    """Test push locally with 'latest' as version."""
+
+    @pytest.mark.parametrize("component_type", ITEM_TYPES)
+    def test_command(self, component_type):
+        """Run the test."""
+        item_name = f"my_{component_type}"
+        self.scaffold_item(component_type, item_name)
```
Comment (marcofavorito) [inline tests/test_cli/test_push.py:223]: what could be used here instead of scaffolding an item? Maybe using `dummy_aea` as agent?

Comment (DavidMinarsch) [reply tests/test_cli/test_push.py:223]: This is fine.

File: aea/cli/push.py [hunk 2]
```diff
@@ -125,10 +123,11 @@ def _check_package_public_id(source_path, item_type, item_id) -> None:
...
     item_name = config.get("name", "")
     item_version = config.get("version", "")
+    actual_item_id = PublicId(item_author, item_name, item_version)
     if (
-        item_id.name != item_name
-        or item_id.author != item_author
-        or item_id.version != item_version
+        actual_item_id.same_prefix(item_id)
+        and not item_id.latest
+        and item_id.version != item_version
```
Comment (DavidMinarsch) [inline aea/cli/push.py:130]: This looks wrong..(we want to raise if the package specified in command and package config *don't* match)

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**